server/repository/migrations/postgres/2021-12-10T10-00_number_table/up.sql

```diff
@@ -1,17 +1,20 @@
 CREATE TYPE number_type AS ENUM (
-    'INBOUND_SHIPMENT',
-    'OUTBOUND_SHIPMENT',
-    'INVENTORY_ADJUSTMENT',
-    'STOCKTAKE',
-    'REQUEST_REQUISITION',
-    'RESPONSE_REQUISITION'
+  'INBOUND_SHIPMENT',
+  'OUTBOUND_SHIPMENT',
+  'INVENTORY_ADJUSTMENT',
+  'STOCKTAKE',
+  'REQUEST_REQUISITION',
+  'RESPONSE_REQUISITION'
 );
 
 -- Numbering table holding a list of typed counters
-CREATE TABLE number (
+CREATE TABLE
+  number (
     id TEXT NOT NULL PRIMARY KEY,
     -- current counter value
     value BIGINT NOT NULL,
     store_id TEXT NOT NULL REFERENCES store(id),
     type number_type NOT NULL
-)
\ No newline at end of file
+  );
+
+CREATE UNIQUE INDEX ix_number_store_type_unique ON number(store_id, type);
```

Comment:
Is this for performance or for uniqueness? Might be bad for performance because there are only a handful of rows and I can imagine a full table scan is faster...
Or is it for "ON CONFLICT DO NOTHING" to work?

Reply:
This is for uniqueness. Hopefully the database engine will still pick a tablescan if it's the fastest option.